doc/DHCPv4_Gateway/DHCPv4_gateway.md

```diff
@@ -0,0 +1,133 @@
+2. Support parsing and assignment of subnets from minigraph/json to config_db. 
+3. Support specifying non-secondary interfaces' gateway address to command line arguments to /usr/sbin/dhcrelay as -g.
+4. isc-dhcp/dhcrelay to support '-g gateway' argument, by porting an existing patch.
+
+### Architecture Design 
+
+This HLD doesn't propose any sonic architectural changes. It follows the existing architecture of sonic - open source asc-dhcp is used for its dhcpv4 requirements.
+This HLD prposes to enhance the dhcpv4 capability by avoiding ambiguity in dhcpv4 packets wrt giaddr field.
+
+### High-Level Design 
+
+This section covers the high level design of the feature/enhancement. This section covers the following points in detail.
+		
+	- Is it a built-in SONiC feature or a SONiC Application Extension? - Built in feature.
+	- What are the modules and sub-modules that are modified for this design? - isc-dhcp.
+	- What are the repositories that would be changed? - sonic-subnet/sonic-buildimage.
+	- Module/sub-module interfaces and dependencies. 
+	- SWSS and Syncd changes in detail - N/A.
+	- DB and Schema changes (APP_DB, ASIC_DB, COUNTERS_DB, LOGLEVEL_DB, CONFIG_DB, STATE_DB) - CONFIG_DB sonic-vlan/sonic-vlan/VLAN_INTERFACE.
+	- Sequence diagram if required. - N/A.
+	- Linux dependencies and interface - N/A.
+	- Warm reboot requirements/dependencies - N/A.
+	- Fastboot requirements/dependencies - N/A.
+	- Scalability and performance requirements/impact - N/A.
+	- Memory requirements - N/A.
+	- Docker dependency - N/A,
+	- Build dependency if any - N/A.
+	- Management interfaces - SNMP, CLI, RestAPI, etc.,
+	- Serviceability and Debug (logging, counters, trace etc) related design
+	- Is this change specific to any platform? Are there dependencies for platforms to implement anything to make this feature work? If yes, explain in detail and inform community in advance. - N/A.
+	- SAI API requirements, CLI requirements, ConfigDB requirements. Design is covered in following sections. - ConfigDB requirement is captured below.
+
+### SAI API 
+
+No change in SAI API.
+
+### Configuration and management 
+(This section should have sub-sections for all types of configuration and management related design. Example sub-sections for "CLI" and "Config DB" are given below. Sub-sections related to data models (YANG, REST, gNMI, etc.,) should be added as required.)
+
+#### Minigraph 
+
+<<Redis DB Structure | Json Style>>
+{
+"VLAN_INTERFACE": {
+	"Vlan1000": {},
+	"Vlan1000|100.1.1.1/27": {"secondary": "true"},
+	"Vlan1000|192.168.0.1/21": {},
+}
+}
+<<End Json Style>>
+
+#### Manifest (if the feature is an Application Extension)
+
+N/A
+
+
+#### CLI/YANG model Enhancements 
+
+Extend CLI:
+
+	$ sudo config interface ip add eth0 20.11.12.13/27 20.11.12.254 --secondary
+	Usage: config interface ip add <interface_name> <ip_addr> <default gateway IP address> <secondary>
+
+Extend yang (sonic-vlan.yang):
+	list VLAN_INTERFACE_IPPREFIX_LIST {
+
+				leaf secondary	{
```

Comment:
Can we extend this support for all other supported L3 interfaces in DHCP relay?

Reply:
This is not in scope for current design. It can be extended in future per the usecase.